src/modules/launcher/Wox.Infrastructure/Storage/StoragePowerToysVersionInfo.cs

```diff
@@ -40,13 +40,29 @@ public static bool Lessthan(string version1, string version2)
             string[] split1 = version1.Split(new string[] { version, period }, StringSplitOptions.RemoveEmptyEntries);
             string[] split2 = version2.Split(new string[] { version, period }, StringSplitOptions.RemoveEmptyEntries);
 
+            // If an incomplete file write resulted in the version number not being saved completely, then the cache must be deleted
+            if (split1.Length != split2.Length || split1.Length != versionLength)
+            {
+                return true;
+            }
+
```

Comment:
I was wondering if we should be even more aggressive by checking if the string matches the regex 
`v[0-9]*\.[0-9]*\.[0-9]*`.  If there is a situation where the file is corrupted somehow then we would even catch that. Thoughts?

Reply:
Wouldn't that be captured by the `TryParse` statements in the following line? I don't think there any possible exceptions with these checks even with custom text as arguments, let me know if I'm missing some cases.

Reply:
Yes, you're right. It would get captured in the TryParse stage.